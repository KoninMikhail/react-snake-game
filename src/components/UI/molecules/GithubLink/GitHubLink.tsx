import { GridItem } from 'components/UI/extend/GridItem/GridItem';
import { ExternalLink } from 'components/UI/atoms/body/ExternalLink/ExternalLink';
import { Text } from 'components/UI/atoms/body/Text/Text';
import { ExternalBorder } from 'assets/borders/ExternalBorder';
import { ExtraShadow } from 'assets/shadows/ExtraShadow';

import { StyledGithubLinkGrid, StyledGitHubIcon } from './StyledGithubLinkGrid';

/**
 * @typedef IGitHubLinkProps
 *
 * @property {String} uri - GitHub URI of the resource being
 * @property {String} label - Link label
 */
export interface IGitHubLinkProps {
  uri: string;
  label: string;
}

/**
 * GitHubLink
 *
 * @description Use for GitHub link on page
 *
 * @param {IGitHubLinkProps} props - component props
 * @returns {JSX.Element}
 */

export const GitHubLink = (props: IGitHubLinkProps): JSX.Element => {
  return (
    <>
      <ExternalLink href={props.uri}>
        <ExtraShadow>
          <ExternalBorder>
            <StyledGithubLinkGrid columns="auto max-content" gap={'1rem'} align="center">
              <GridItem align="center">
                <Text as="span">{props.label}</Text>
              </GridItem>
              <GridItem align="center">
                <StyledGitHubIcon />
              </GridItem>
            </StyledGithubLinkGrid>
          </ExternalBorder>
        </ExtraShadow>
      </ExternalLink>
    </>
  );
};
