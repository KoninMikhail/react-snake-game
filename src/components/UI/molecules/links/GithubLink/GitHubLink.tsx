import { GridItem } from 'components/UI/extend/GridItem/GridItem';
import { ExternalLink } from 'components/UI/atoms/body/ExternalLink/ExternalLink';
import { Text } from 'components/UI/atoms/body/Text/Text';
import { ExternalBorder } from 'components/UI/atoms/assets/borders/ExternalBorder';
import { ExtraShadow } from 'components/UI/atoms/assets/shadows/ExtraShadow';

import { StyledGithubLinkGrid, StyledGitHubIcon } from './StyledGithubLinkGrid';

/**
 * @typedef IGitHubLinkProps
 *
 * @property {String} uri - GitHub URI of the resource being
 * @property {String} label - Link label
 */
export interface IGitHubLinkProps {
  uri: string;
  label: string;
}

/**
 * GitHubLink
 *
 * Use for GitHub link on page
 *
 * @param {IGitHubLinkProps} props - component props
 * @returns {JSX.Element}
 */
export const GitHubLink = ({ uri, label }: IGitHubLinkProps) => {
  return (
    <ExternalLink href={uri}>
      <ExtraShadow>
        <ExternalBorder>
          <StyledGithubLinkGrid columns="auto max-content" gap={'1rem'} align="center">
            <GridItem align="center">
              <Text as="span">{label}</Text>
            </GridItem>
            <GridItem align="center">
              <StyledGitHubIcon />
            </GridItem>
          </StyledGithubLinkGrid>
        </ExternalBorder>
      </ExtraShadow>
    </ExternalLink>
  );
};
