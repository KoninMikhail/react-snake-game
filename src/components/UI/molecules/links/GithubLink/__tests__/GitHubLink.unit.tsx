import { render, screen } from '@testing-library/react';
import { GitHubLink } from '../GitHubLink';

describe('Github link', () => {
  const uri = 'https://github.com';
  const label = 'welcome to github';

  it('render element with props', () => {
    render(<GitHubLink uri={uri} label={label} />);

    expect(screen.getByText(label)).toBeInTheDocument();
    expect(screen.getByRole('link')).toHaveAttribute('href', uri);
  });

  it('render correctly', () => {
    const { container } = render(<GitHubLink uri={uri} label={label} />);
    expect(container).toMatchSnapshot();
  });
});
