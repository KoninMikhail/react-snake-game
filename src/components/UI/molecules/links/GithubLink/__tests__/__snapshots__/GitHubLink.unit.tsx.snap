// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Github link render correctly 1`] = `
.c5 {
  display: block;
  position: relative;
  grid-column: auto;
  grid-row: auto;
  justify-self: auto;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c0 {
  cursor: pointer;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.3s ease-out;
  transition: all 0.3s ease-out;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c6 {
  font-size: 14px;
  font-family: 'pixeloid_bold','Segoe UI','Roboto','Oxygen','Ubuntu','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  font-weight: normal;
  color: white;
}

.c2 {
  border: 3px dashed deeppink;
}

.c1 {
  box-shadow: rgba(0,0,0,0.25) 0px 54px 55px,rgba(0,0,0,0.12) 0px -12px 30px, rgba(0,0,0,0.12) 0px 4px 6px,rgba(0,0,0,0.17) 0px 12px 13px, rgba(0,0,0,0.09) 0px -3px 5px;
}

.c3 {
  display: grid;
  position: relative;
  grid-template-columns: auto max-content;
  grid-template-rows: 1fr;
  grid-gap: 1rem;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  padding: 0.8rem 0.8rem 0.8rem;
  background: black;
}

.c7 {
  fill: white;
  padding-top: 3px;
}

@media (min-width:1200px) {
  .c6 {
    font-size: 1vw;
  }
}

@media (min-width:1200px) {
  .c4 {
    padding: 0.8vw 0.8vw 0.8vw;
  }
}

@media (min-width:1200px) {
  .c7 {
    width: 1.5em;
    height: 1.5em;
  }
}

<div>
  <a
    class="c0"
    href="https://github.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3 c4"
        >
          <div
            class="c5"
          >
            <span
              class="c6"
            >
              welcome to github
            </span>
          </div>
          <div
            class="c5"
          >
            <div
              class="c7"
            />
          </div>
        </div>
      </div>
    </div>
  </a>
</div>
`;
